### PR TITLE
feat: [IOCOM-868] Update SEND message header and content to the new design system

### PR DESCRIPTION
**Pull request by adelloste**
## Short description
This PR updates the header and content of a SEND message.

_NOTE: Other PRs will follow to update the rest of the message._

<details open><summary>Details</summary>
<p>

| message details |
| - |
|  |

</p>
</details> 

## List of changes proposed in this pull request
- added `MessageDetailHeader`
- updated `MessageDetailsContent`
- added tests

## How to test
Using the `io-dev-api-server`, generate a SEND message. Enable the design system and navigate to the message detail. Check that the content is displayed correctly.

### CODE REVIEW

File: ts/features/pn/screens/LegacyMessageDetailsScreen.tsx [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+  startPaymentStatusTracking,
+  updatePaymentForMessage
+} from "../store/actions";
+import { GlobalState } from "../../../store/reducers/types";
+import { selectedPaymentIdSelector } from "../store/reducers/payments";
+import { InfoScreenComponent } from "../../../components/infoScreen/InfoScreenComponent";
+import { renderInfoRasterImage } from "../../../components/infoScreen/imageRendering";
+import genericErrorIcon from "../../../../img/wallet/errors/generic-error-icon.png";
+
+export type MessageDetailsScreenNavigationParams = Readonly<{
```
Comment (Vangaorth) [inline ts/features/pn/screens/LegacyMessageDetailsScreen.tsx:43]: Should we remove this type and rely on the new screen only?

File: ts/features/pn/screens/MessageDetailsScreen.tsx
Comment (Vangaorth) [inline ts/features/pn/screens/MessageDetailsScreen.tsx:1]: If I'm not wrong, we should remove the BaseScreenComponent and the SafeAreView component and rely only on the useSecondHeader hook to handle the screen and its parameters

Comment (Vangaorth) [inline ts/features/pn/screens/MessageDetailsScreen.tsx:1]: InfoSCreenComponent has been replaced by OperationResultScreenComponent. By replacing that, we also get rid of renderInfoRasterImage and genericErrorIcon

File: ts/features/pn/components/MessageDetails.tsx [hunk 2]
```diff
@@ -1,193 +1,27 @@
-import React, { useCallback, createRef, useRef } from "react";
-import { pipe } from "fp-ts/lib/function";
-import * as O from "fp-ts/lib/Option";
-import * as RA from "fp-ts/lib/ReadonlyArray";
-import * as SEP from "fp-ts/lib/Separated";
-import { View } from "react-native";
+import React from "react";
 import { ScrollView } from "react-native-gesture-handler";
```
Comment (Vangaorth) [inline ts/features/pn/components/MessageDetails.tsx:2]: Is this import correct or should we use the react-native one?

File: ts/features/messages/components/MessageDetail/__tests__/MessageDetailHeader.test.tsx [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import React, { ComponentProps } from "react";
+import { render } from "@testing-library/react-native";
+import { MessageDetailHeader } from "../MessageDetailHeader";
+
+const defaultProps: ComponentProps<typeof MessageDetailHeader> = {
+  subject: "Subject",
+  createdAt: new Date("2021-10-18T16:00:30.541Z")
+};
+
+describe("MessageDetailHeader component", () => {
```
Comment (Vangaorth) [inline ts/features/messages/components/MessageDetail/__tests__/MessageDetailHeader.test.tsx:10]: Should we also add a snapshot with all properties?

File: ts/features/pn/components/MessageDetails.tsx
Comment (Vangaorth) [inline ts/features/pn/components/MessageDetails.tsx:1]: What about a snapshot test on this one?

File: ts/features/pn/components/__test__/MessageDetailsContent.test.tsx [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import React from "react";
+import { render } from "@testing-library/react-native";
+import { MessageDetailsContent } from "../MessageDetailsContent";
+
+describe("MessageDetailsContent component", () => {
+  it("should match the snapshot when abstract is defined", () => {
```
Comment (Vangaorth) [inline ts/features/pn/components/__test__/MessageDetailsContent.test.tsx:6]: Should we add a snapshot test with undefined abstract?

File: ts/features/pn/screens/MessageDetailsScreen.tsx
Comment (Vangaorth) [inline ts/features/pn/screens/MessageDetailsScreen.tsx:1]: Should we add a snapshot test for this screen?

**Review by Vangaorth — CHANGES_REQUESTED** [review]

File: ts/features/pn/screens/LegacyMessageDetailsScreen.tsx (hunk 1 above)
Comment (adelloste) [reply ts/features/pn/screens/LegacyMessageDetailsScreen.tsx]: Applied https://github.com/pagopa/io-app/pull/5431/commits/1bee02ac06a5964aa3d0497341ec972fc7cb1d14

File: ts/features/pn/screens/MessageDetailsScreen.tsx
Comment (adelloste) [reply ts/features/pn/screens/MessageDetailsScreen.tsx:1]: Applied https://github.com/pagopa/io-app/pull/5431/commits/983145b745415b798a6c98c63d0e592b7bebc34f

File: ts/features/pn/components/MessageDetails.tsx (hunk 2 above)
Comment (adelloste) [reply ts/features/pn/components/MessageDetails.tsx]: Applied https://github.com/pagopa/io-app/pull/5431/commits/8eb6bb8ecaad487899b7ff08094c09e55a552b60

File: ts/features/pn/components/__test__/MessageDetailsContent.test.tsx (hunk 4 above)
Comment (adelloste) [reply ts/features/pn/components/__test__/MessageDetailsContent.test.tsx:6]: Applied https://github.com/pagopa/io-app/pull/5431/commits/3efbf44c589ca4c843609d0a82613db12aec9439

File: ts/features/pn/components/MessageDetails.tsx
Comment (adelloste) [reply ts/features/pn/components/MessageDetails.tsx:1]: Applied https://github.com/pagopa/io-app/pull/5431/commits/c1099cbe1a2e3009587212477a81dddfd225d157

**Review by Vangaorth — APPROVED** [review]
LGTM

**Pull request merged**